### PR TITLE
Remove the `WorkflowRun/index.jelly` file

### PR DESCRIPTION
**Pull request by janfaracik**
Relates to https://github.com/jenkinsci/jenkins/pull/8110

This PR removes the `WorkflowRun/index.jelly` file as if **#8110** is merged, it'd be possible to rely on the `Run/index.jelly` file from Jenkins instead. 

It would be preferable to have plugins extend pages rather than to reimplement them to make changes in core easier and more consistent. This will be important with upcoming works on the Jenkins user experience, such as introducing app bars, new widgets and adjusting the sidebar.

If there's anything I've missed do let me know.

**Before**
image

**After**
image

### Submitter checklist

- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by timja** [comment]
What is that failed to determine thing?

**Comment by jglick** [comment]
Marking as draft since this cannot be merged as is—only if and when the core PR is merged, and the baseline version here is updated accordingly.

**Comment by jglick** [comment]
> What is that failed to determine thing?

Something about changesets. https://github.com/jenkinsci/jenkins/pull/8110#issuecomment-1584362971

**Comment by jglick** [comment]
Set the `jenkins.version` here to an incremental build of the specific revision of the core PR it matches.

**Comment by jglick** [comment]
Does this still need work to adapt to https://github.com/jenkinsci/jenkins/pull/8110/commits/f4057971eb3e0f7c47200c8053ceac6c8aaaec15?

File: pom.xml [hunk 1]
```diff
@@ -63,7 +63,7 @@
     </pluginRepositories>
     <properties>
         <changelist>999999-SNAPSHOT</changelist>
-        <jenkins.version>2.361.4</jenkins.version>
+        <jenkins.version>2.410-rc33839.8e8205848340</jenkins.version>
```
Comment (jglick) [inline pom.xml:66]: https://github.com/jenkinsci/workflow-job-plugin/pull/352#discussion_r1226878951 by the way

Comment (jglick) [reply pom.xml:66]: Now out of date. To pick up https://github.com/jenkinsci/jenkins/pull/8110/commits/16408d859a4ef8a095cb96c6cbea51427f926d90 (and https://github.com/jenkinsci/jenkins/pull/8110/commits/c862d9bed1f5080b931396901df99a34a0419a0e, not that it should matter)
```suggestion
        <jenkins.version>2.415-rc33995.c862d9b_ed1f5</jenkins.version>
```

**Comment by janfaracik** [comment]
Before
Screenshot 2024-05-17 at 11 19 13

After
Screenshot 2024-05-17 at 11 17 49

Tried it out with a basic pipeline, looks to be working fine. Only one issue has been raised for the core PR and that's  https://issues.jenkins.io/browse/JENKINS-73168

**Comment by jglick** [comment]
OK, just mark ready for review whenever you think it is safe to merge.

**Comment by jglick** [comment]
(Requesting review from reviewers of the core PR, since they probably more context for this than I.)

**Review by timja — APPROVED** [review]
Thanks, LGTM, I tested this at the time

**Pull request merged**